exopy_hqc_legacy/instruments/drivers/visa/agilent_pna.py

```diff
@@ -394,7 +394,7 @@ def sweep_x_axis(self):
                 'SENSe{}:FREQuency:STARt?'.format(self._channel)))*1e-9
             sweep_stop = float(self._pna.query(
                 'SENSe{}:FREQuency:STOP?'.format(self._channel)))*1e-9
-            return np.linspace(sweep_start, sweep_stop, sweep_points)
+            return np.linspace(sweep_start, sweep_stop, int(sweep_points))
```

Comment:
I would suggest doing `int(round(x))` if you suspect a float as otherwise int will floor the number.